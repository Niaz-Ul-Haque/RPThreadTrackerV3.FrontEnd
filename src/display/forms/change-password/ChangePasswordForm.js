import React from 'react';
import PropTypes from 'prop-types';
import { AvField } from 'availity-reactstrap-validation';
import { FormGroup, Col, Label } from 'reactstrap';
import validator from './_validator';

const propTypes = {
	handleInputChange: PropTypes.func.isRequired
};

const ChangePasswordForm = (props) => {
	const {
		handleInputChange
	} = props;
	return (
		<div>
			<FormGroup row>
				<Col xs="12" lg="3">
					<Label
						data-spec="current-password-label"
						htmlFor="current-password"
					>
						Current Password:
					</Label>
				</Col>
				<Col xs="12" lg="9">
					<AvField
						data-spec="current-password-field"
						name="currentPassword"
						placeholder="Current Password"
						type="password"
						onChange={handleInputChange}
						validate={validator.currentPassword}
					/>
				</Col>
			</FormGroup>
			<FormGroup row>
				<Col xs="12" lg="3">
					<Label htmlFor="new-password" data-spec="new-password-label">New Password:</Label>
				</Col>
				<Col xs="12" lg="9">
					<AvField
						data-spec="new-password-field"
						name="newPassword"
						placeholder="New Password"
						type="password"
						onChange={handleInputChange}
						validate={validator.newPassword}
					/>
				</Col>
			</FormGroup>
			<FormGroup row>
				<Col xs="12" lg="3">
					<Label
						htmlFor="confirm-new-password"
						data-spec="confirm-new-password-label"
					>
						Confirm New Password:
					</Label>
				</Col>
				<Col xs="12" lg="9">
					<AvField
						data-spec="confirm-new-password-field"
						name="confirmNewPassword"
						placeholder="Confirm New Password"
						type="password"
						onChange={handleInputChange}
						validate={validator.confirmNewPassword}
					/>
				</Col>
			</FormGroup>
		</div>
	);
};

ChangePasswordForm.propTypes = propTypes;
export default ChangePasswordForm;
