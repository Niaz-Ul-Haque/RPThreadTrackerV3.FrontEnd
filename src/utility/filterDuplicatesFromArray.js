const filterDuplicatesFromArray = (array) => {
	return array.filter((elem, pos, arr) => elem && arr.indexOf(elem) == pos);
}
export default filterDuplicatesFromArray;
