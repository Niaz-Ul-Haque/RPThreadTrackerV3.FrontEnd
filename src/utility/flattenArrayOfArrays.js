const flattenArrayOfArrays = array => array.reduce((a, b) => a.concat(
	Array.isArray(b) ? flattenArrayOfArrays(b) : b
), []);
export default flattenArrayOfArrays;
