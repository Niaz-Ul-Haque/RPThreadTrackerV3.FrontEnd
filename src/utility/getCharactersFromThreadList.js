import { filterDuplicatesFromArray } from './';

const getCharactersFromThreadList = (threads) => {
	if (!threads || threads.length < 1) {
		return [];
	}
	const characters = threads.map(t => t.thread.character);
	return filterDuplicatesFromArray(characters, 'characterId');
};
export default getCharactersFromThreadList;
