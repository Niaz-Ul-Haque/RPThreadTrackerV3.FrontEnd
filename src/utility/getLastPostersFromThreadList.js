import { filterDuplicatesFromArray } from './';

const getLastPostersFromThreadList = (threads) => {
	if (!threads || threads.length < 1) {
		return [];
	}
	const lastPosters = threads.map((t) => {
		if (!t.status) {
			return '';
		}
		return t.status.LastPosterUrlIdentifier;
	});
	return filterDuplicatesFromArray(lastPosters);
};
export default getLastPostersFromThreadList;
