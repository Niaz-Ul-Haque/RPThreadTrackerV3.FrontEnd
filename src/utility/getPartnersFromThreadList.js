import { filterDuplicatesFromArray } from './';

const getPartnersFromThreadList = (threads) => {
	if (!threads || threads.length < 1) {
		return [];
	}
	const partners = threads.map((t) => {
		if (!t.thread) {
			return '';
		}
		return t.thread.partnerUrlIdentifier;
	});
	return filterDuplicatesFromArray(partners);
};
export default getPartnersFromThreadList;
