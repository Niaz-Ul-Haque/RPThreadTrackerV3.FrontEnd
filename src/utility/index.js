export { default as getQuery } from './getQuery';
export { default as flattenArrayOfArrays } from './flattenArrayOfArrays';
export { default as filterDuplicatesFromArray } from './filterDuplicatesFromArray';
export { default as getCharactersFromThreadList } from './getCharactersFromThreadList';
export { default as getPartnersFromThreadList } from './getPartnersFromThreadList';
export { default as getLastPostersFromThreadList } from './getLastPostersFromThreadList';
export { default as getTagsFromThreadList } from './getTagsFromThreadList';
