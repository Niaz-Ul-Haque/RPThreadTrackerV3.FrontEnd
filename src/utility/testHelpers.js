export const DATA_SPEC_ATTRIBUTE_NAME = 'data-spec';

export const getSpecWrapper = (componentWrapper, specName) => {
	const specWrappers = componentWrapper.find(`[${DATA_SPEC_ATTRIBUTE_NAME}="${specName}"]`);
	return specWrappers;
};
