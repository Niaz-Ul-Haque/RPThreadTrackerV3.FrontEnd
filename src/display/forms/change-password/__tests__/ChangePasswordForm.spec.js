// #region imports
import React from 'react';
import { shallow } from 'enzyme';
import { getSpecWrapper } from '../../../../utility/testHelpers';
import ChangePasswordForm from '../ChangePasswordForm';
// #endregion imports

const createTestProps = propOverrides => ({
	handleInputChange: jest.fn(),
	...propOverrides
});

describe('rendering', () => {
	it('should render valid snapshot', () => {
		const props = createTestProps();
		const jsx = (<ChangePasswordForm {...props} />);
		const element = shallow(jsx);
		expect(element).toMatchSnapshot();
	});
});

describe('behavior', () => {
	it('should handle input change for current password', () => {
		const handleInputChange = jest.fn();
		const props = createTestProps({ handleInputChange });
		const jsx = (<ChangePasswordForm {...props} />);
		const element = shallow(jsx);
		const field = getSpecWrapper(element, 'current-password-field');
		field.simulate('change');
		expect(handleInputChange).toHaveBeenCalledTimes(1);
	});
	it('should handle input change for new password', () => {
		const handleInputChange = jest.fn();
		const props = createTestProps({ handleInputChange });
		const jsx = (<ChangePasswordForm {...props} />);
		const element = shallow(jsx);
		const field = getSpecWrapper(element, 'new-password-field');
		field.simulate('change');
		expect(handleInputChange).toHaveBeenCalledTimes(1);
	});
	it('should handle input change for confirm new password', () => {
		const handleInputChange = jest.fn();
		const props = createTestProps({ handleInputChange });
		const jsx = (<ChangePasswordForm {...props} />);
		const element = shallow(jsx);
		const field = getSpecWrapper(element, 'confirm-new-password-field');
		field.simulate('change');
		expect(handleInputChange).toHaveBeenCalledTimes(1);
	});
});
