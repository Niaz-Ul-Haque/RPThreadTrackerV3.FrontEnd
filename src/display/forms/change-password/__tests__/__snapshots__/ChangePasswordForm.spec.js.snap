// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`rendering snapshots should render valid snapshot 1`] = `
<div>
  <FormGroup
    row={true}
    tag="div"
  >
    <Col
      lg="3"
      tag="div"
      widths={
        Array [
          "xs",
          "sm",
          "md",
          "lg",
          "xl",
        ]
      }
      xs="12"
    >
      <Label
        htmlFor="current-password"
        tag="label"
        widths={
          Array [
            "xs",
            "sm",
            "md",
            "lg",
            "xl",
          ]
        }
      >
        Current Password:
      </Label>
    </Col>
    <Col
      lg="9"
      tag="div"
      widths={
        Array [
          "xs",
          "sm",
          "md",
          "lg",
          "xl",
        ]
      }
      xs="12"
    >
      <AvField
        data-spec="current-password-field"
        name="currentPassword"
        onChange={[MockFunction]}
        placeholder="Current Password"
        type="password"
        validate={
          Object {
            "required": Object {
              "errorMessage": "You must enter your current password.",
              "value": true,
            },
          }
        }
      />
    </Col>
  </FormGroup>
  <FormGroup
    row={true}
    tag="div"
  >
    <Col
      lg="3"
      tag="div"
      widths={
        Array [
          "xs",
          "sm",
          "md",
          "lg",
          "xl",
        ]
      }
      xs="12"
    >
      <Label
        htmlFor="new-password"
        tag="label"
        widths={
          Array [
            "xs",
            "sm",
            "md",
            "lg",
            "xl",
          ]
        }
      >
        New Password:
      </Label>
    </Col>
    <Col
      lg="9"
      tag="div"
      widths={
        Array [
          "xs",
          "sm",
          "md",
          "lg",
          "xl",
        ]
      }
      xs="12"
    >
      <Tooltip
        align={
          Object {
            "offset": Array [
              0,
              10,
            ],
          }
        }
        arrowContent={null}
        data-spec="newPassword-tooltip"
        destroyTooltipOnHide={false}
        mouseEnterDelay={0}
        mouseLeaveDelay={0.1}
        overlay="Password must contain an uppercase character, a lowercase character, a numeric digit, and a symbol."
        overlayStyle={
          Object {
            "width": 300,
          }
        }
        placement="top"
        prefixCls="rc-tooltip"
        trigger={
          Array [
            "hover",
          ]
        }
      >
        <AvField
          data-spec="new-password-field"
          name="newPassword"
          onBlur={[MockFunction]}
          onChange={[MockFunction]}
          onFocus={[MockFunction]}
          placeholder="Password"
          type="password"
          validate={
            Object {
              "minLength": Object {
                "errorMessage": "Your password must be longer than 6 characters.",
                "value": 6,
              },
              "required": Object {
                "errorMessage": "You must enter a new password.",
                "value": true,
              },
            }
          }
        />
      </Tooltip>
    </Col>
  </FormGroup>
  <FormGroup
    row={true}
    tag="div"
  >
    <Col
      lg="3"
      tag="div"
      widths={
        Array [
          "xs",
          "sm",
          "md",
          "lg",
          "xl",
        ]
      }
      xs="12"
    >
      <Label
        htmlFor="confirm-new-password"
        tag="label"
        widths={
          Array [
            "xs",
            "sm",
            "md",
            "lg",
            "xl",
          ]
        }
      >
        Confirm New Password:
      </Label>
    </Col>
    <Col
      lg="9"
      tag="div"
      widths={
        Array [
          "xs",
          "sm",
          "md",
          "lg",
          "xl",
        ]
      }
      xs="12"
    >
      <AvField
        data-spec="confirm-new-password-field"
        name="confirmNewPassword"
        onChange={[MockFunction]}
        placeholder="Confirm New Password"
        type="password"
        validate={
          Object {
            "match": Object {
              "errorMessage": "Your passwords must match.",
              "value": "newPassword",
            },
            "required": Object {
              "errorMessage": "You must confirm your password.",
              "value": true,
            },
          }
        }
      />
    </Col>
  </FormGroup>
</div>
`;
